core/src/main/java/org/infinispan/eviction/EvictionManager.java

```diff
@@ -23,5 +22,5 @@
     * Handles notifications of evicted entries
     * @param evicted The entries that were just evicted
     */
-   void onEntryEviction(Map<? extends K, InternalCacheEntry<? extends K, ? extends V>> evicted);
+   void onEntryEviction(Map<K, Map.Entry<K, V>> evicted);
```

Comment:
Do you really need this change? I thought we were moving to single-entry eviction notification, and then metadata should probably be included in the event.

Reply:
I want to move to single in a separate PR. This is just to get the generics to work temporarily :D

Reply:
Ok, I didn't realize there was any problem with generics here :)